### PR TITLE
feat(kit,nuxt,schema,vite,webpack): migrate to nitropack v3 nightly

### PR DESCRIPTION
**Pull request by danielroe**
### 📚 Description

This migrates to a particular version of nitro v3 and h3 v2 nightly releases, for further testing.

- [x] https://github.com/unjs/nitro/pull/2546
- [x] consider using `nitro` rather than `nitropack` (at least internally)
- [x] reimplement nitro appConfig using vfs
- [x] resolve type issues (upstream?)
- [x] consider augmenting both `nitro/types` and `nitropack/types`

More significant breaking changes are coming in `h3`, along with the following further todos

- [ ] update Nuxt upgrade guide for nitro-related changes
- [ ] update nuxt/test-utils in a forward/backward compatible way
- [ ] remove internal dependencies on `event.node`
- [ ] remove db0 in Nitro upstream
- [ ] refactor runtimeConfig inside Nuxt

### CODE REVIEW

File: packages/schema/src/config/typescript.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export default defineUntypedSchema({
       $resolve: (val) => {
         const defaults = [
           // Nitro auto-imported/augmented dependencies
-          'nitropack',
+          // TODO: add 'nitro/types'
```
Comment (danielroe) [inline packages/schema/src/config/typescript.ts:37]: need to add subpath export support for typescript.hoist

File: packages/nuxt/src/core/runtime/nitro/app-config.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  if (!event) {
+    return _sharedAppConfig
+  }
+  // Reuse cached app config from event context
+  if (event.context.nitro.appConfig) {
+    return event.context.nitro.appConfig
+  }
+  // Prepare app config for event context
+  const appConfig = klona(_inlineAppConfig)
+  event.context.nitro.appConfig = appConfig
```
Comment (pi0) [inline packages/nuxt/src/core/runtime/nitro/app-config.ts:20]: Even for polyfill/backward compatibility, filling nitro context in event would be misleading we should store it in context.nuxt

File: packages/vite/src/server.ts [hunk 3]
```diff
@@ -104,7 +104,7 @@ export async function buildServer (ctx: ViteBuildContext) {
   } satisfies vite.InlineConfig, ctx.nuxt.options.vite.$server || {}))
 
   if (!ctx.nuxt.options.dev) {
-    const nitroDependencies = await tryResolveModule('nitropack/package.json', ctx.nuxt.options.modulesDir)
+    const nitroDependencies = await tryResolveModule('nitro/package.json', ctx.nuxt.options.modulesDir)
```
Comment (pi0) [inline packages/vite/src/server.ts:107]: This is available now via `nitro/runtime/meta`

File: packages/webpack/src/presets/base.ts [hunk 4]
```diff
@@ -125,7 +125,7 @@ function baseAlias (ctx: WebpackConfigContext) {
     ...ctx.alias,
   }
   if (ctx.isClient) {
-    ctx.alias['#internal/nitro'] = resolve(ctx.nuxt.options.buildDir, 'nitro.client.mjs')
+    ctx.alias['nitro/runtime'] = resolve(ctx.nuxt.options.buildDir, 'nitro.client.mjs')
```
Comment (pi0) [inline packages/webpack/src/presets/base.ts:128]: Just checking it was because of runtime config access right?

File: test/bundle.test.ts [hunk 5]
```diff
@@ -86,6 +87,7 @@ describe.skipIf(process.env.SKIP_BUNDLE_SIZE === 'true' || process.env.ECOSYSTEM
         "@unhead/dom",
         "@unhead/shared",
         "@unhead/ssr",
+        "db0",
```
Comment (pi0) [inline test/bundle.test.ts:90]: It shouldn't be added by default in the prod bundle seems strange!

File: packages/nuxt/src/core/plugins/import-protection.ts [hunk 6]
```diff
@@ -85,6 +85,7 @@ export const ImportProtectionPlugin = createUnplugin(function (options: ImportPr
         matched = true
       }
       if (matched) {
+        console.log(id, cache[id])
```
Comment (pi0) [inline packages/nuxt/src/core/plugins/import-protection.ts:88]: (i guess leftover?)

```suggestion
```

File: packages/webpack/src/presets/base.ts (hunk 4 above)
Comment (danielroe) [reply packages/webpack/src/presets/base.ts:128]: Exactly. This is so we can use isomorphic code that gets replaced with different runtime config code on client.

Comment (pi0) [reply packages/webpack/src/presets/base.ts:128]: Okay! I have an idea to get rid of it!

File: packages/nuxt/src/core/plugins/import-protection.ts (hunk 6 above)
Comment (danielroe) [reply packages/nuxt/src/core/plugins/import-protection.ts]: Thank you 🙏

File: packages/webpack/src/presets/base.ts (hunk 4 above)
Comment (danielroe) [reply packages/webpack/src/presets/base.ts:128]: 🙌

**Pull request merged**